activerecord/lib/active_record/table_metadata.rb

```diff
@@ -11,11 +11,7 @@ def initialize(klass, arel_table, reflection = nil)
     end
 
     def arel_attribute(column_name)
-      if klass
-        klass.arel_attribute(column_name, arel_table)
-      else
-        arel_table[column_name]
-      end
+      arel_table[column_name]
```

Comment:
:thinking: What's the reason to not remove this method at all?

Reply:
Actually `arel_attribute` other than this method is unused in the internal, but that was only way to resolve attribute aliases for a long time, and I (and probably someone) had oftenly used the method in app code, so I suppose a deprecation (as a separate commit/PR) is more user-friendly than removing the method directly.

Reply:
Or, do you mean that just expose `arel_table` as public?

Reply:
`arel_attribute` is private API as well if I understand it well. Within this patch is just just "shortcut" to `arel_table`. The same method is in `core.rb` and `relation.rb`. I was looking for opportunity to remove `arel_attribute` method at all and replace all calls with `arel_table` directly (since it is the only thing `arel_attribute` method do now).

Reply:
Even if it is an internal API, deprecation may be preferable to removing it directly if it is widely used.

Ref #28293, #28351.

Anyway, I've pushed the removing commit 27af39a5588f706d58fc8c68932b2f386b604eb7.

Reply:
I did quick search and this method is used in few projects. I think the biggest public one are https://github.com/ManageIQ/manageiq and https://github.com/composite-primary-keys/composite_primary_keys. I can send PR to fix CPK (they do versioning per Rails version, it should not cause troubles since new Rails is not usable with old gem version on their side). Or is it worth it deprecation?

Reply:
I also use `arel_attribute` in a PR for 3rd party gem https://github.com/kufu/activerecord-bitemporal/pull/62, since it was only way to resolve attribute aliases even if it is private API, as I said at the above.

I suppose it is worth deprecation.